controllers/update_status.go

```
@@ -330,7 +330,7 @@ func tryConnectionOptions(logger logr.Logger, cluster *fdbv1beta2.FoundationDBCl
 		logger.Error(err, "Error getting connection string from cluster", "connectionString", connectionString)
 	}
 
-	return originalConnectionString, nil
+	return originalConnectionString, fmt.Errorf("could not get connection string from cluster")
```

I would not return here the `originalConnectionString` for defensive programming reasons: caller should not be allowed to ignore the error and use the connection string anyways.

---

Done, looks like nothing is using it in the error case.